### PR TITLE
tests: made func tests a bit flexible

### PR DESCRIPTION
**Pull request by lmesz**
With this minimal fix in tests, syslog-ng can be installed on arbitary place in a host. This place can be passed via SYSLOG_NG_BINARY and functional test can be executed like before.

### CODE REVIEW

**Comment by bazsi** [comment]
I am not against such a change, but we originally intended to run the functest in the build environment.  In fact this is nice as the installed version can be tested easily. 

I think the module-path calculation needs to be changed as well though.

**Comment by lmesz** [comment]
Thanks for the fast response and info.
I understand the base concept, and I didn't want to go againt that, but I would like to write func test for one of the module in incubator and didn't want to copy or reimplement the test infrastructure.

And thanks modify the determination of module path.

**Comment by lmesz** [comment]
Hello,

Based on your suggestion changed the module path also.

**Comment by bazsi** [comment]
Almost there. 
It would be better to name get_modules () as get_module_path() or something.

Also, the installation path shouldn't be hard coded in the test script. The syslog-ng binary should tell it to us. It should be added to --version or --module-registry

And last but not least the installed modules should only be used iff we are not building intree. As if I have syslog-ng on my system where I also develop it, it's load the system supplied one instead of my locally compiled one

**Comment by lmesz** [comment]
I made some modifications that you mentioned before. One remained I will try to fix it on tomorrow.

File: lib/plugin.c [hunk 1]
```diff
@@ -481,6 +481,12 @@ plugin_free_candidate_modules(GlobalConfig *cfg)
 }
 
 void
+plugin_get_module_path(FILE *out)
+{
+    fprintf(out, "Module path: %s\n", module_path);
+}
+
```
Comment (bazsi) [inline lib/plugin.c]: well, if it is called a getter it should be a getter and not printing it.

Comment (bazsi) [inline lib/plugin.c]: also, the module_path global is pretty sure to be at the wrong location in mainloop.c and should probably be pushed down to plugin.c, but that's a separate issue.

File: tests/functional/globals.py [hunk 2]
```diff
@@ -1,13 +1,27 @@
 import os
 
+def get_builtin_modules():
+    module_path = os.popen("%s --version | grep \"Module Path\" | cut -d ' ' -f 3" % get_syslog_ng_binary(), 'r').read().split(".")
+    return '%s:' % module_path
+
```
Comment (bazsi) [inline tests/functional/globals.py]: while reviewing the patches I see two separate interpretation of the term "module_path" which should be separated:

1) it is used for the default module directory as used by "make install"
2) it is used to store the list of directories where modules are searched for.

The 2nd usually includes the first, but probably the name should indicate which one we are working on:

"module_dir" should be used for installation directory
"module_path" should only be used for the colon separated list of directories.

File: tests/functional/globals.py [hunk 3]
```diff
@@ -1,13 +1,27 @@
 import os
 
+def get_builtin_modules():
+    module_path = os.popen("%s --version | grep \"Module Path\" | cut -d ' ' -f 3" % get_syslog_ng_binary(), 'r').read().split(".")
+    return '%s:' % module_path
```
Comment (bazsi) [inline tests/functional/globals.py]: We should probably use a dash in the name of "Module-Path", or rather Module-Directory and also a colon to keep the rfc822 style formatting of --version output.

Comment (lmesz) [inline tests/functional/globals.py]: Fixed it and changed from "Module Path" to "Module-Directory:"

File: lib/plugin.c (hunk 1 above)
Comment (lmesz) [inline lib/plugin.c]: Should I open an issue about that ?

File: tests/functional/globals.py (hunk 2 above)
Comment (lmesz) [inline tests/functional/globals.py]: Okay sorry yes that's true. I fixed the names based on your  instructions. From now on I guess it is clear which functions would do what.

**Comment by bazsi** [comment]
Hi,

I have refined your patches on my branch named lmesz/remove-fix-relative-path-of-syslog-ng-in-func-test

Please let me know if you are fine with the changes and if you are pls modify your pull request.

Cheers,
Bazsi

**Comment by lmesz** [comment]
Thanks for the fixups, I modified my branch based on what you did.

**Comment by lmesz** [comment]
Sorry rebased and fixed how to find module path in version info

**Comment by bazsi** [comment]
The python part looks now ok, but the C part is still missing some of the pieces. If you want I can prepare a branch that contains both your code and mine. What do you think?

**Comment by lmesz** [comment]
Of course it would be good. Thank you.
My goal at the end of the PR is to use this func-test ecosystem to test a module in incubator.
Thanks your help again.

**Comment by bazsi** [comment]
this should be closed without merging once #858 is merged.

**Comment by bazsi** [comment]
I think I am closing this instead to clear up clutter.

**Pull request closed without merging**